README.rst

```diff
@@ -151,10 +151,36 @@ General
     logger = logging.getLogger("cli.logger")
     logger.info('hi')
 
-**Notes**: - All loggers set up and created before the initialization of the Cloud Foundry logging library will
-be left untouched. - When using Flask and Sanic with the logging library a before and
+**Notes**: All loggers set up and created before the initialization of the Cloud Foundry logging library will
+be left untouched. When using Flask and Sanic with the logging library a before and
 after request middleware is attached, and it will capture response times for each request.
 
+Setting and getting correlation ID
+""""""""""""""""""""""""""""""""""
+
+When using cf_logging in a web application you don't need to care about setting and getting correlation ID,
+because the logging frameworks supported will fetch it from HTTP headers and will set it to the log output for you.
+But if you are developing an application triggered via cli or messaging or by schedule, you might want to set the correlation ID manually,
+so all log entries can be filtered lately based on this correlation_id.
+
+Setting and getting correlation_id can be done via:
+
+.. code:: python
+
+    cf_logging.FRAMEWORK.context.get_correlation_id()
+    cf_logging.FRAMEWORK.context.set_correlation_id(value)
+
+The context created in cf_logging that have been initialized when no specific framework is used (like ``cf_logging.init()``) is global which means that setting correlation_id will result in same correlatino_id for the all logs from this point on.
```

Comment:
Whenever a correlation id is set after initializing cf_logging without a specific framework (ex: ``cf_logging.init()``) - the correlation id set would be used for each log.

If you need *a thread .... and provide your* own thread safe context implementation.